entities_service/service/security.py

```diff
@@ -50,6 +53,111 @@ async def get_openid_config() -> OpenIDConfiguration:
         raise ValueError("Could not parse OpenID configuration.") from exc
 
 
+async def verify_user_access_token(token: str) -> tuple[bool, int | None, str | None]:
+    """Verify a user-provided GitLab access token."""
+    # Get current user
+    async with AsyncClient(headers={"Authorization": f"Bearer {token}"}) as client:
```

Comment:
async with AsyncClient on L94 : suggestion: this request to GitLab API can be reused.

Reply:
The request is performed within the `async with` block. The two requests are different. I could create the client and reuse it - but I prefer using a context for HTTP requests to ensure any HTTP transports are properly closed.
The alternative is to call the `aclose()` method after usage.

Reply:
Or maybe I could just initialize it as a variable and then use that in the contexts?

Reply:
Hmm, after reading more in [the documentation](https://www.python-httpx.org/async/#opening-and-closing-clients), I think maybe going for `aclose()` and a single client instance is the way to go.